### ISSUE TITLE
How does `macro inherited` work with generics?

### ISSUE THREAD

**Issue by rmosolgo** [crystal-lang/crystal#1793]
`inherited` works nicely in a non-generic case:

```crystal
# $ crystal eval
class Parent
  macro inherited
    @@class_array = [1,2,3]
  end
end

class Child < Parent
  @@class_array << 4
  puts @@class_array
end
# => [1, 2, 3, 4]
```

But in a generic case, it fails: 

```crystal
# $ crystal eval
class GenericParent(T)
  macro inherited
    @@class_array = [1,2,3]
  end
end

class GenericChild < GenericParent(Int32)
  @@class_array << 4
  puts @@class_array
end
# Error in line 8: undefined method '<<' for Nil
```

I expect it to work the same as the non-generic case. Am I missing something? Or is this a bug?

**Comment by asterite** [comment]
A bug, inherited doesn't work right now with generic types.

**Comment by asterite** [comment]
Closed in favor of #2665
